test/state/system_contracts.cpp

```diff
@@ -72,4 +94,44 @@ StateDiff system_call(const StateView& state_view, const BlockInfo& block,
     // TODO: Should we return empty diff if no system contracts?
     return state.build_diff(rev);
 }
+
+std::pair<StateDiff, std::vector<Requests>> system_call_block_end(const StateView& state_view,
+    const BlockInfo& block, const state::BlockHashes& block_hashes, evmc_revision rev, evmc::VM& vm)
+{
+    State state{state_view};
+    std::vector<Requests> requests;
+    for (const auto& [since, addr, get_input, request_type] : SYSTEM_CONTRACTS_BLOCK_END)
```

Comment:
Actually maybe it makes senst not to duplicate the loop code, and reuse the same, I'll try it